### PR TITLE
Creates git repository for git user

### PR DESCRIPTION
**Pull request by SahilKalra98**
Part 2 of issue: #188

### CODE REVIEW

File: src/api/mapping.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ mappingRouter.post('/', (req, res) => {
       }
     ]
   }
-  const projectPath = '~/projects'
+  const projectPath = '/home/git'
   const scriptPath = path.join(__dirname, '../../scripts')
   exec(`
     mkdir -p ${projectPath}
```
Comment (songz) [inline src/api/mapping.ts]: gotta switch user to git

File: src/api/mapping.ts [hunk 2]
```diff
@@ -3,14 +3,13 @@ import express from 'express'
...
 import { Mapping } from '../types/general'
 const mappingRouter = express.Router()
 const exec = util.promisify(cp.exec)
-const getNextPort = (map, start=3002):number =>{
-  if(!map[start]) return start
-  if(map[start]) start +=1 
+const getNextPort = (map, start = 3002): number => {
+  if (!map[start]) return start
+  if (map[start]) start += 1
   return getNextPort(map, start)
```
Comment (SahilKalra98) [inline src/api/mapping.ts]: From Lint - autofix

File: src/api/mapping.ts [hunk 3]
```diff
@@ -19,8 +18,8 @@ mappingRouter.post('/', (req, res) => {
   const map = domainKeys.reduce((acc, e) => {
     acc[e.port] = true
     return acc
-  }, {}) 
-  const portCounter = getNextPort(map) 
+  }, {})
+  const portCounter = getNextPort(map)
```
Comment (SahilKalra98) [inline src/api/mapping.ts]: From Lint - autofix

File: src/api/mapping.ts [hunk 4]
```diff
@@ -47,22 +46,28 @@ mappingRouter.post('/', (req, res) => {
...
-    cd ${projectPath}/${fullDomain}
-    git config user.email "root@ipaddress"
-    git config user.name "user"
-    echo 'module.exports = ${JSON.stringify(prodConfig)}' > deploy.config.js
-    git add .
-    git commit -m "Initial Commit"`).then(() => {
+  const projectPath = '/home/git'
+  const scriptPath = '.scripts'
+  exec('id -u git').then(result => {
+    exec(
```
Comment (songz) [inline src/api/mapping.ts]: move backticks here: 
```
exec(`
```

File: src/api/mapping.ts [hunk 5]
```diff
@@ -41,32 +49,25 @@ mappingRouter.post('/', (req, res) => {
...
-    const mappingObject: Mapping = {
-      domain: req.body.domain,
-      subDomain: req.body.subDomain,
-      port: req.body.port || `${portCounter}`,
-      ip: req.body.ip || '127.0.0.1',
-      id: uuid4(),
-      gitLink: `git@${req.body.domain}:${projectPath}/${fullDomain}`,
-      fullDomain,
-    }
-    domainKeys.push(mappingObject)
```
Comment (songz) [inline src/api/mapping.ts]: this should be brought back

File: src/types/general.ts [hunk 6]
```diff
@@ -4,8 +4,8 @@ type Mapping = {
   port: string
   ip: string
   id: string
-  gitLink: string
-  fullDomain: string
+  gitLink?: string
+  fullDomain?: string
```
Comment (songz) [inline src/types/general.ts]: no need.

**Pull request merged**